internal/test/environment/clean.go

```diff
@@ -38,6 +38,10 @@ func (e *Execution) Clean(t testingT) {
 	if platform == "linux" {
 		deleteAllPlugins(t, client, e.protectedElements.plugins)
 	}
+
+	if e.DaemonInfo.Swarm.NodeID != "" {
+		deleteAllServices(t, client, platform, e.protectedElements.services)
```

Comment:
Looking at the order in which we remove objects, I suspect we'd have to remove services _before_ we remove volumes and networks (otherwise those will still be in use, and fail to be removed). Likely _before_ we call `deleteAllContainers()` as well, otherwise we'll try to remove containers that are part of a service, resulting in SwarmKit starting _new_ containers (as part of the reconciliation loop).

This may be tricky, because removing a service is asynchronous, so even if the service is removed, the tasks (containers) backing the service may still be there.

@vdemeester any suggestions for that one? Perhaps we need to update `deleteAllContainers()` to filter out containers that are part of a service, or would it work if we already have removed the service first, and then manually remove the containers (because if the service was removed, the reconciliation-loop may no longer be running, so we can safely remove the containers?)